### PR TITLE
add option to automatically start multi-room clean

### PR DESCRIPTION
**Pull request by normen**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -118,6 +118,7 @@ For the underlying communication layer, it uses a port of the no-longer maintain
...
 | `autoroom`        | false         | set to true to generate rooms from robot (only S6) or set to array of room name strings (see semi automatic below)                                                                                                                                                                                                                                                                                                                                                                          |
+| `roomTimeout`     | false         | when set to anything above "0" - will initiate room cleaning for multiple rooms automatically after the given time in seconds                                                                                                                                                                                                                                                                                                                                                               |
```
Comment (afharo) [inline README.md:121]:
```suggestion
| `roomTimeout`     | 0         | when set to anything above "0" - will initiate room cleaning for multiple rooms automatically after the given time in seconds                                                                                                                                                                                                                                                                                                                                                               |
```

File: index.js [hunk 2]
```diff
@@ -113,6 +113,7 @@ class XiaomiRoborockVacuum {
     this.config.findMe = config.findMe || false;
     this.config.findMeWord = config.findMeWord || "where are you";
     this.config.delay = config.delay || false;
+    this.config.roomTimeout = config.roomTimeout == undefined ? 0 : config.roomTimeout;
```
Comment (afharo) [inline index.js:116]: FYI, you can also write this as
```suggestion
    this.config.roomTimeout = config.roomTimeout ?? 0;
```

File: index.js [hunk 3]
```diff
@@ -954,11 +955,23 @@ class XiaomiRoborockVacuum {
...
           `ACT setCleaningRoom | ${this.model} | Disable cleaning Room ID ${roomId}.`
         );
         this.roomIdsToClean.delete(roomId);
+        if (this.config.roomTimeout > 0) {
+          this.log.info(`ACT setCleaningRoom | ${this.model} | Start timeout to clean rooms`);
+          clearTimeout(this._roomTimeout)
+          if (this.roomIdsToClean.size > 0) {
+            this._roomTimeout = setTimeout(this.setCleaning.bind(this, true), this.config.roomTimeout * 1000)
+          }
+        }
```
Comment (afharo) [inline index.js:974]: Do you mind moving this piece of logic to a function and to call it on both branches? If feels like repeating the logic :)

**Review by afharo — COMMENTED** [review]
These changes are great! Thank you @normen!

I've added a couple of suggestions. Would you mind running through them?

File: index.js (hunk 2 above)
Comment (normen) [reply index.js:116]: Doesn't seem to work for me, maybe because I am sporting Node 12?

**Comment by normen** [comment]
there.

**Comment by normen** [comment]
Uhm, I think you released to my branch there :)

**Review by afharo — APPROVED** [review]
LGTM! Thank you for doing these changes! I'll merge it and publish it in a sec!

**Pull request merged**